### DISCUSSION TITLE
Support project.el external roots

### DISCUSSION THREAD

**Discussion by fargiolas** [minad/consult#975] [Ideas]
Probably not so popular in these monorepo days but project.el supports setting one or more external roots to grep and find files into just like if they belonged to the main project. See `project-or-external-find-file` and `project-or-external-find-regexp` .

It would be nice if consult would honor this setting and extend find and grep commands to external roots if set.

This is mostly an enhancement, at the moment I'm getting the desired effect with (this actually discards the dir argument but works for me as I'm using it only for projects anyway):

```elisp
(defun fa/consult-ripgrep (&optional dir initial)
  (interactive "P")
  (let ((dirs (append (list (consult--project-root))
                      (project-external-roots (project-current)))))
    (consult--grep "Ripgrep" #'consult--ripgrep-make-builder dirs initial)))

(defun fa/consult-fd (&optional dir initial)
  (interactive "P")
  (let ((dirs (append (list (consult--project-root))
                      (project-external-roots (project-current)))))
    (consult-fd dirs initial)))
```

**Comment by minad** [comment]
It would be nice to support this. Maybe we should change the `consult-project-function` to either return a directory name or a list of directory names? I am not sure how many changes are needed for this, but patches welcome!
